syntax = "proto3";

package huddle_chat;

import "google/protobuf/timestamp.proto";

message UserForLogin {
  int64 user_id = 1;
  string username = 2;
  string email = 3;
  string avatar = 4;
  int32 online_status = 5;
  google.protobuf.Timestamp created_at = 6;
  string password = 7;
}

message LoginRequest {
  string email = 1;
}

message LoginResponse {
  UserForLogin user = 1;
}

message RegisterRequest {
  string username = 1;
  string password = 2;
  string email = 3;
}

message RegisterResponse {
  bool success = 1;
  string message = 2;
  int32 verification_code = 3;
}

service UserService {
  rpc GetUserForLogin(LoginRequest) returns (LoginResponse) {}
  rpc RegisterUser(RegisterRequest) returns (RegisterResponse) {}
}
